### ISSUE TITLE
Alert rules -> unexpected error when doin anything in the query window

### ISSUE THREAD

**Issue by tjerkkamstra** [grafana/grafana#63401]
**What happened**:
When trying to add a new alert rule, I get an unexpected error whenever I click out of the query window again. Nothing is saved. It doesn't matter whether I enter in a good query or a load full of gibberish.

![UnexpectedError](https://user-images.githubusercontent.com/69219115/219648370-f512e4a8-1363-4568-93e0-1d982bd64e93.gif)

**What you expected to happen**:
When I click next to the query window I can move on and put in a condition and so on. 

**How to reproduce it (as minimally and precisely as possible)**:
1. Go to the add rule window
2. Click in the query window
3. Do anything you want there
4. Click out of it
5. Error appears
**Anything else we need to know?**:
I wondered wheter it was caused by the latest version, so I reversed back to 9.3.4 and got the same error. I tried a few older versions and then on version 9.2.10 it works without errors. I find nothing useful in the logfiles but have attached them anyway.

**Environment**:
- Grafana version: 9.3.6
- Data source type & version: SQLite 3.1.0
- OS Grafana is installed on:  Debian 5.10.162-1 (2023-01-21) running inside Docker version 23.0.1, build a5ee5b1 using the grafana-oss (Alpine versions)
- User OS & Browser: Windows 11 with Firefox 110.0 (64-bits) (But the issue also applies to the other browsers installed, Edge, Brave, Chrome, Chromium)
- Grafana plugins: all standard plugins and SQLite by frser as mentioned above.
- Others:

```
An unexpected error happened
TypeError: e.fields[0] is undefined

di@https://araluen.tucker.nl/grafana/public/build/AlertingRuleForm.d245068ee1365350e124.js:145:386
div
de@https://araluen.tucker.nl/grafana/public/build/805.dab4bcdab5a814dbcdba.js:2:92533
vi@https://araluen.tucker.nl/grafana/public/build/AlertingRuleForm.d245068ee1365350e124.js:252:158
div
de@https://araluen.tucker.nl/grafana/public/build/805.dab4bcdab5a814dbcdba.js:2:92533
fieldset
d@https://araluen.tucker.nl/grafana/public/build/2362.5e93872490cc5d80351e.js:1406:1760
div
div
o@https://araluen.tucker.nl/grafana/public/build/2362.5e93872490cc5d80351e.js:6335:172
No@https://araluen.tucker.nl/grafana/public/build/AlertingRuleForm.d245068ee1365350e124.js:292:2635
div
div
div
t@https://araluen.tucker.nl/grafana/public/build/805.dab4bcdab5a814dbcdba.js:2:711588
p@https://araluen.tucker.nl/grafana/public/build/2362.5e93872490cc5d80351e.js:424:3558
div
form
R@https://araluen.tucker.nl/grafana/public/build/805.dab4bcdab5a814dbcdba.js:2:927680
$o@https://araluen.tucker.nl/grafana/public/build/AlertingRuleForm.d245068ee1365350e124.js:292:5853
div
37417/b.Contents@https://araluen.tucker.nl/grafana/public/build/2362.5e93872490cc5d80351e.js:5799:2272
div
div
div
t@https://araluen.tucker.nl/grafana/public/build/805.dab4bcdab5a814dbcdba.js:2:711588
p@https://araluen.tucker.nl/grafana/public/build/2362.5e93872490cc5d80351e.js:424:3558
div
b@https://araluen.tucker.nl/grafana/public/build/2362.5e93872490cc5d80351e.js:5799:1289
t@https://araluen.tucker.nl/grafana/public/build/9076.637f4fd32666ef6824c7.js:1:211
7328/Vo<@https://araluen.tucker.nl/grafana/public/build/AlertingRuleForm.d245068ee1365350e124.js:313:914
c@https://araluen.tucker.nl/grafana/public/build/2362.5e93872490cc5d80351e.js:1207:301
u@https://araluen.tucker.nl/grafana/public/build/2362.5e93872490cc5d80351e.js:1207:855
WithErrorBoundary
Suspense
c@https://araluen.tucker.nl/grafana/public/build/2362.5e93872490cc5d80351e.js:1207:301
_c@https://araluen.tucker.nl/grafana/public/build/2362.5e93872490cc5d80351e.js:8831:350
t@https://araluen.tucker.nl/grafana/public/build/805.dab4bcdab5a814dbcdba.js:2:1060860
t@https://araluen.tucker.nl/grafana/public/build/805.dab4bcdab5a814dbcdba.js:2:1062881
main
Cd@https://araluen.tucker.nl/grafana/public/build/2362.5e93872490cc5d80351e.js:8792:6825
t@https://araluen.tucker.nl/grafana/public/build/805.dab4bcdab5a814dbcdba.js:2:1057478
div
o@https://araluen.tucker.nl/grafana/public/build/2362.5e93872490cc5d80351e.js:1996:4308
19175/t.KBarProvider@https://araluen.tucker.nl/grafana/public/build/805.dab4bcdab5a814dbcdba.js:2:271052
l@https://araluen.tucker.nl/grafana/public/build/2362.5e93872490cc5d80351e.js:6143:19549
c@https://araluen.tucker.nl/grafana/public/build/2362.5e93872490cc5d80351e.js:1207:301
u@https://araluen.tucker.nl/grafana/public/build/2362.5e93872490cc5d80351e.js:1207:855
l@https://araluen.tucker.nl/grafana/public/build/805.dab4bcdab5a814dbcdba.js:2:1040510
Su@https://araluen.tucker.nl/grafana/public/build/2362.5e93872490cc5d80351e.js:8844:115
```

```
logger=ngalert.api t=2023-02-17T11:43:02.211775064Z level=debug msg="user has no access to any namespaces"
logger=user.service t=2023-02-17T11:43:05.434122648Z level=debug msg="got user from local cache" cachekey=signed-in-user-1-1
logger=accesscontrol.service t=2023-02-17T11:43:05.434183615Z level=debug msg="using cached permissions" key=rbac-permissions-1-user-1
logger=user.service t=2023-02-17T11:43:05.434122665Z level=debug msg="got user from local cache" cachekey=signed-in-user-1-1
logger=accesscontrol.service t=2023-02-17T11:43:05.434385535Z level=debug msg="using cached permissions" key=rbac-permissions-1-user-1
logger=ngalert.api t=2023-02-17T11:43:05.434759586Z level=debug msg="user does not have access to any namespaces"
logger=ngalert.api t=2023-02-17T11:43:05.435160612Z level=debug msg="user has no access to any namespaces"
logger=ngalert.scheduler t=2023-02-17T11:43:10.001098957Z level=debug msg="Alert rules fetched" rulesCount=0 foldersCount=0
logger=ngalert.state.manager t=2023-02-17T11:43:11.61024184Z level=debug msg="Recording state cache metrics" now=2023-02-17T11:43:11.610231371Z
logger=ngalert.scheduler t=2023-02-17T11:43:20.000605031Z level=debug msg="Alert rules fetched" rulesCount=0 foldersCount=0
logger=accesscontrol.service t=2023-02-17T11:43:25.443713583Z level=debug msg="fetch permissions from store" key=rbac-permissions-1-user-1
logger=accesscontrol.service t=2023-02-17T11:43:25.443789201Z level=debug msg="fetch permissions from store" key=rbac-permissions-1-user-1
logger=accesscontrol.service t=2023-02-17T11:43:25.444862514Z level=debug msg="cache permissions" key=rbac-permissions-1-user-1
logger=accesscontrol.service t=2023-02-17T11:43:25.445114666Z level=debug msg="cache permissions" key=rbac-permissions-1-user-1
logger=ngalert.api t=2023-02-17T11:43:25.445486142Z level=debug msg="user has no access to any namespaces"
logger=ngalert.api t=2023-02-17T11:43:25.445633357Z level=debug msg="user does not have access to any namespaces"
logger=secrets t=2023-02-17T11:43:26.524248094Z level=debug msg="Removing expired data keys from cache..."
logger=secrets t=2023-02-17T11:43:26.524296394Z level=debug msg="Removing expired data keys from cache finished successfully"
logger=ngalert.state.manager t=2023-02-17T11:43:26.609509443Z level=debug msg="Recording state cache metrics" now=2023-02-17T11:43:26.609500708Z
logger=ngalert.sender.router t=2023-02-17T11:43:26.612039107Z level=debug msg="Attempting to sync admin configs" count=0
logger=ngalert.sender.router t=2023-02-17T11:43:26.612071559Z level=debug msg="Finish of admin configuration sync"
logger=ngalert.multiorg.alertmanager t=2023-02-17T11:43:26.613989317Z level=debug msg="synchronizing Alertmanagers for orgs"
logger=alertmanager org=1 t=2023-02-17T11:43:26.614872918Z level=debug msg="neither config nor template have changed, skipping configuration sync."
logger=ngalert.multiorg.alertmanager t=2023-02-17T11:43:26.615442732Z level=debug msg="done synchronizing Alertmanagers for orgs"
logger=user.service t=2023-02-17T11:43:28.616497627Z level=debug msg="got user from local cache" cachekey=signed-in-user-1-1
logger=accesscontrol.service t=2023-02-17T11:43:28.616559338Z level=debug msg="using cached permissions" key=rbac-permissions-1-user-1
logger=user.service t=2023-02-17T11:43:28.616909723Z level=debug msg="got user from local cache" cachekey=signed-in-user-1-1
logger=accesscontrol.service t=2023-02-17T11:43:28.616955904Z level=debug msg="using cached permissions" key=rbac-permissions-1-user-1
logger=user.service t=2023-02-17T11:43:28.638171808Z level=debug msg="got user from local cache" cachekey=signed-in-user-1-1
logger=accesscontrol.service t=2023-02-17T11:43:28.638216788Z level=debug msg="using cached permissions" key=rbac-permissions-1-user-1
logger=ngalert.api t=2023-02-17T11:43:28.638713912Z level=debug msg="user has no access to any namespaces"
logger=ngalert.scheduler t=2023-02-17T11:43:30.000914098Z level=debug msg="Alert rules fetched" rulesCount=0 foldersCount=0
logger=accesscontrol.service t=2023-02-17T11:43:32.361923479Z level=debug msg="using cached permissions" key=rbac-permissions-1-user-1
logger=accesscontrol.evaluator t=2023-02-17T11:43:32.36223943Z level=debug msg="matched scope" userscope=datasources:* targetscope=datasources:uid:4d6eyLcVz
logger=datasources t=2023-02-17T11:43:32.362306679Z level=debug msg="Querying for data source via SQL store" uid=4d6eyLcVz orgId=1
logger=secrets.kvstore t=2023-02-17T11:43:32.363433005Z level=debug msg="got secret value" orgId=1 type=datasource namespace=Klimalogg
logger=plugin.frser-sqlite-datasource t=2023-02-17T11:43:32.363999289Z level=debug msg="Received request for data"
logger=plugin.frser-sqlite-datasource t=2023-02-17T11:43:32.364048201Z level=debug msg="Variables replaced"
logger=plugin.frser-sqlite-datasource t=2023-02-17T11:43:32.364091338Z level=debug msg="Macros applied"
logger=plugin.frser-sqlite-datasource t=2023-02-17T11:43:32.3644034Z level=debug msg="Fetched data from database"
logger=plugin.frser-sqlite-datasource t=2023-02-17T11:43:32.364460869Z level=debug msg="Finished query" refID=A
logger=ngalert.scheduler t=2023-02-17T11:43:40.00076241Z level=debug msg="Alert rules fetched" rulesCount=0 foldersCount=0
logger=ngalert.state.manager t=2023-02-17T11:43:41.610211059Z level=debug msg="Recording state cache metrics" now=2023-02-17T11:43:41.610201249Z
logger=ngalert.scheduler t=2023-02-17T11:43:50.001176091Z level=debug msg="Alert rules fetched" rulesCount=0 foldersCount=0
logger=ngalert.state.manager t=2023-02-17T11:43:56.609668439Z level=debug msg="Recording state cache metrics" now=2023-02-17T11:43:56.609659081Z
logger=ngalert.scheduler t=2023-02-17T11:44:00.000693174Z level=debug msg="Alert rules fetched" rulesCount=0 foldersCount=0
logger=ngalert.scheduler t=2023-02-17T11:44:10.00066522Z level=debug msg="Alert rules fetched" rulesCount=0 foldersCount=0
logger=ngalert.state.manager t=2023-02-17T11:44:11.610138011Z level=debug msg="Recording state cache metrics" now=2023-02-17T11:44:11.610126698Z
logger=ngalert.scheduler t=2023-02-17T11:44:20.000685732Z level=debug msg="Alert rules fetched" rulesCount=0 foldersCount=0
logger=accesscontrol.service t=2023-02-17T11:44:21.23110817Z level=debug msg="fetch permissions from store" key=rbac-permissions-1-user-1
logger=accesscontrol.service t=2023-02-17T11:44:21.231156222Z level=debug msg="fetch permissions from store" key=rbac-permissions-1-user-1
logger=accesscontrol.service t=2023-02-17T11:44:21.232209882Z level=debug msg="cache permissions" key=rbac-permissions-1-user-1
logger=accesscontrol.service t=2023-02-17T11:44:21.232346715Z level=debug msg="cache permissions" key=rbac-permissions-1-user-1
logger=ngalert.api t=2023-02-17T11:44:21.232829137Z level=debug msg="user has no access to any namespaces"
logger=ngalert.api t=2023-02-17T11:44:21.23289478Z level=debug msg="user does not have access to any namespaces"
logger=secrets t=2023-02-17T11:44:26.524242477Z level=debug msg="Removing expired data keys from cache..."
logger=secrets t=2023-02-17T11:44:26.524290435Z level=debug msg="Removing expired data keys from cache finished successfully"
logger=ngalert.state.manager t=2023-02-17T11:44:26.609501545Z level=debug msg="Recording state cache metrics" now=2023-02-17T11:44:26.609493228Z
logger=ngalert.sender.router t=2023-02-17T11:44:26.612935346Z level=debug msg="Attempting to sync admin configs" count=0
logger=ngalert.sender.router t=2023-02-17T11:44:26.612968542Z level=debug msg="Finish of admin configuration sync"
logger=ngalert.multiorg.alertmanager t=2023-02-17T11:44:26.615926745Z level=debug msg="synchronizing Alertmanagers for orgs"
logger=alertmanager org=1 t=2023-02-17T11:44:26.616772099Z level=debug msg="neither config nor template have changed, skipping configuration sync."
logger=ngalert.multiorg.alertmanager t=2023-02-17T11:44:26.617320739Z level=debug msg="done synchronizing Alertmanagers for orgs"
logger=ngalert.scheduler t=2023-02-17T11:44:30.000661293Z level=debug msg="Alert rules fetched" rulesCount=0 foldersCount=0
logger=accesscontrol.service t=2023-02-17T11:44:39.497961167Z level=debug msg="fetch permissions from store" key=rbac-permissions-1-user-1
logger=accesscontrol.service t=2023-02-17T11:44:39.497977778Z level=debug msg="fetch permissions from store" key=rbac-permissions-1-user-1
logger=accesscontrol.service t=2023-02-17T11:44:39.498997628Z level=debug msg="cache permissions" key=rbac-permissions-1-user-1
logger=accesscontrol.service t=2023-02-17T11:44:39.499312408Z level=debug msg="cache permissions" key=rbac-permissions-1-user-1
logger=user.service t=2023-02-17T11:44:39.518115014Z level=debug msg="got user from local cache" cachekey=signed-in-user-1-1
logger=accesscontrol.service t=2023-02-17T11:44:39.518164091Z level=debug msg="using cached permissions" key=rbac-permissions-1-user-1
logger=ngalert.api t=2023-02-17T11:44:39.518673573Z level=debug msg="user has no access to any namespaces"
logger=ngalert.scheduler t=2023-02-17T11:44:40.000544568Z level=debug msg="Alert rules fetched" rulesCount=0 foldersCount=0
logger=ngalert.state.manager t=2023-02-17T11:44:41.610040333Z level=debug msg="Recording state cache metrics" now=2023-02-17T11:44:41.610030293Z
logger=user.service t=2023-02-17T11:44:42.759186736Z level=debug msg="got user from local cache" cachekey=signed-in-user-1-1
logger=accesscontrol.service t=2023-02-17T11:44:42.759244152Z level=debug msg="using cached permissions" key=rbac-permissions-1-user-1
logger=accesscontrol.evaluator t=2023-02-17T11:44:42.759543778Z level=debug msg="matched scope" userscope=datasources:* targetscope=datasources:uid:4d6eyLcVz
logger=datasources t=2023-02-17T11:44:42.759612598Z level=debug msg="Querying for data source via SQL store" uid=4d6eyLcVz orgId=1
logger=secrets.kvstore t=2023-02-17T11:44:42.760779991Z level=debug msg="got secret value" orgId=1 type=datasource namespace=Klimalogg
logger=plugin.frser-sqlite-datasource t=2023-02-17T11:44:42.761362891Z level=debug msg="Received request for data"
logger=plugin.frser-sqlite-datasource t=2023-02-17T11:44:42.761440057Z level=debug msg="Variables replaced"
logger=plugin.frser-sqlite-datasource t=2023-02-17T11:44:42.761465698Z level=debug msg="Macros applied"
logger=plugin.frser-sqlite-datasource t=2023-02-17T11:44:42.761757873Z level=debug msg="Fetched data from database"
logger=plugin.frser-sqlite-datasource t=2023-02-17T11:44:42.761801101Z level=debug msg="Finished query" refID=A
logger=ngalert.scheduler t=2023-02-17T11:44:50.001232948Z level=debug msg="Alert rules fetched" rulesCount=0 foldersCount=0
logger=ngalert.state.manager t=2023-02-17T11:44:56.610446274Z level=debug msg="Recording state cache metrics" now=2023-02-17T11:44:56.610435506Z
logger=ngalert.scheduler t=2023-02-17T11:45:00.000651748Z level=debug msg="Alert rules fetched" rulesCount=0 foldersCount=0
logger=ngalert.scheduler t=2023-02-17T11:45:10.000675736Z level=debug msg="Alert rules fetched" rulesCount=0 foldersCount=0
logger=ngalert.state.manager t=2023-02-17T11:45:11.610074504Z level=debug msg="Recording state cache metrics" now=2023-02-17T11:45:11.610065724Z
logger=ngalert.scheduler t=2023-02-17T11:45:20.000696033Z level=debug msg="Alert rules fetched" rulesCount=0 foldersCount=0
logger=secrets t=2023-02-17T11:45:26.524303152Z level=debug msg="Removing expired data keys from cache..."
logger=secrets t=2023-02-17T11:45:26.524346188Z level=debug msg="Removing expired data keys from cache finished successfully"
logger=ngalert.state.manager t=2023-02-17T11:45:26.609548322Z level=debug msg="Recording state cache metrics" now=2023-02-17T11:45:26.609540842Z
logger=ngalert.sender.router t=2023-02-17T11:45:26.614055095Z level=debug msg="Attempting to sync admin configs" count=0
logger=ngalert.sender.router t=2023-02-17T11:45:26.614088589Z level=debug msg="Finish of admin configuration sync"
logger=ngalert.multiorg.alertmanager t=2023-02-17T11:45:26.618022905Z level=debug msg="synchronizing Alertmanagers for orgs"
logger=alertmanager org=1 t=2023-02-17T11:45:26.61887491Z level=debug msg="neither config nor template have changed, skipping configuration sync."
logger=ngalert.multiorg.alertmanager t=2023-02-17T11:45:26.619430648Z level=debug msg="done synchronizing Alertmanagers for orgs"
logger=ngalert.scheduler t=2023-02-17T11:45:30.000610279Z level=debug msg="Alert rules fetched" rulesCount=0 foldersCount=0
logger=ngalert.scheduler t=2023-02-17T11:45:40.001465165Z level=debug msg="Alert rules fetched" rulesCount=0 foldersCount=0
logger=ngalert.state.manager t=2023-02-17T11:45:41.610320507Z level=debug msg="Recording state cache metrics" now=2023-02-17T11:45:41.610311829Z
```

**Comment by gillesdemey** [comment]
Hi! Have you tried running the latest 9.4.0 release? It includes a fix for handling empty data frames that yielded a similar exception.

**Comment by tjerkkamstra** [comment]
Hi Giles,

Well no, because as far as I can see there is no container for that version
yet.

Do we know for sure it solves this issue as well? Because i could not find
it in the bug base.

**Comment by gillesdemey** [comment]
I've checked and can confirm that the issue does not occur in 9.4 but does happen in 9.3.x – the PR that resolves it is https://github.com/grafana/grafana/pull/61814 and we should back-port this one to the previous minor release.

**Comment by gillesdemey** [comment]
I've back-ported the fix to 9.3.x – you can either upgrade to the next 9.4.0 minor version or the 9.3.x patch release when either of those are released.

I'm going to go ahead and close this issue – feel free to re-open the issue if it still persists after upgrading.
